src/extension.ts

```diff
@@ -93,10 +93,26 @@ export function activate(context: vscode.ExtensionContext) {
 
       // Highlight matching tags
       tagStyler.clearDecorations()
-      editor.selections
-        .map(sel => findMatchingTag(tagsList, editor.document.offsetAt(sel.active)))
-        .filter(match => match && (match.opening !== match.closing || config.highlightSelfClosing))
-        .forEach(match => tagStyler.decoratePair(match as hmt.Match, editor))
+
+      let matches = []
+      if (!config.highlightFromContent) {
+        matches = editor.selections
+          .map(sel => findMatchingTag(tagsList, editor.document.offsetAt(sel.active)))
+          .filter(
+            match => match && (match.opening !== match.closing || config.highlightSelfClosing)
+          )
+      } else {
+        matches = editor.selections.map(
+          sel =>
+            getTagsForPosition(tagsList, editor.document.offsetAt(sel.active))
+              .filter(
+                match => match && (match.opening !== match.closing || config.highlightSelfClosing)
```

Comment:
In this case, match is always guaranteed to exist, because no matches, would simply be an empty array, so checking if match exists is redundant.

Would probably be cleaner if this whole: `getTagsForPosition(...).filter().slice()` part was a function something like: `getTagForPosition` in tagMatcher. Apparently I skipped out on tests for that matching, but it would be nice if at least this part had some small test cases - at least a "basic" one, a "nothing found" one and a "self closing tag" one. Let me know if you want to add them.

Reply:
I've added a `getTagForPosition` function as requested. I just repurposed 2 of the existing tests for the new function, but you can add more if that is not sufficient.